### DISCUSSION TITLE
Require password change at login for newly created users (also, emailing credentials)

### DISCUSSION THREAD

**Discussion by SteveMCIW** [Blazam-App/BLAZAM#299] [Ideas]
It would be great if templates would allow us to force the user to change their password at first login. Also, if it was possible to send the new user's initial logon details to them via email, that would help with the workflow too. (I'm assuming that neither of these are possible at the moment, but feel free to let me know if they are!)

**Comment by jacobsen9026** [comment]
This is a great suggestion that honestly, should already have been added.
I actually forgot about it, as at my workplace we have other methods of forcing a password change prior to logon.

The email notification will have to wait for a future release, but I can get the force password change into the next update 0.9.2

I'll have to postpone the 0.9.2 release from today to tomorrow or Thursday to have enough time to test and validate.

**Reply by jacobsen9026** [reply]
0.9.2 just landed, includes the Require Password Change in templates.

**Comment by jacobsen9026** [comment]
> It would be great if templates would allow us to force the user to change their password at first login. Also, if it was possible to send the new user's initial logon details to them via email, that would help with the workflow too. (I'm assuming that neither of these are possible at the moment, but feel free to let me know if they are!)

Hey @SteveMCIW, in your opinion, is it better to just notify on all "permissable" ou's or to force registration to the desired ou's (which is obviously more complicated). Both were planned to be included, but knowing which you would would prefer first can help me with my roadmap.

My plan was to make notification registration be similar to action assignment in the permissions access levels. Choose an OU, then the notification actions per AD object type desired. Let me know if you have any thoughts on that.

If I don't hear back, I will go with both approaches.

Thank you.

**Reply by SteveMCIW** [reply]
Hi @jacobsen9026  - I'm not 100% sure what you're referring to here? Is it some kind of alerting when actions are carried out on AD objects? (i.e. so managers or monitoring systems can pick up when changes are made? If so, then that's a cool feature, albeit not one that I've a pressing need for in our system.

**Comment by jacobsen9026** [comment]
@SteveMCIW  I just realized I've been ranting about something completely different than you're asking for...

> possible to send the new user's initial logon details to them via email

Are we talking a "personal" email, one not generated via the email field in AD.... (I would assume so as you would need said credentials to log in)? Then yes, very very easily... I could add a send details via email button on the final "Create User" confirmation page (which needs a bit of work tbh) and ask for an email address to send to... does that fulfill your workflow, feel free to suggest anything else.

**Reply by jacobsen9026** [reply]
Also, if you have an opinion, how much customization of the email message would you like, everything from styling/themes to content and formats.

**Reply by SteveMCIW** [reply]
In our setup, our users have their own email addresses (we provide services to external companies), so the email account is independent of our Active Directory. We would use the email field in their new user object as a reference to send their initial login details to. Perhaps have it configurable, so you could pick to default to either that, or specify an alternative email address to send to?

**Reply by jacobsen9026** [reply]
I've been trying to think of ways to improve your suggestion, but I think you pretty much nailed it.

I was originally hesitant about the setting to ask for an alternate email (thinking it can be automated with logic), but after going over every workflow type I can think of, it's the only one that can cover all use cases.

For user creation templates that are not marked to automatically send an email out, I still plan to have a button that will be available to send one on the confirmation page.

I'm going to pass on email template customization for now, as that is a nice to have but not necessary.

In a future update, after the above is implemented, I plan to build out the notifications system to allow users to subscribe to events based on OU for things like new user creation, password changes, group assignments, etc.

**Reply by jacobsen9026** [reply]
@jacobsen9026 Note to self: Include warning on template edit page when send email is true and ask for alternate is false but email field is set

**Reply by jacobsen9026** [reply]
@jacobsen9026 Also require email field if not already required and send email is true while ask for alternate is false

**Reply by jacobsen9026** [reply]
Hmm, didn't notify here, but just clarifying, require password change is fully implemented and functional in the most recent update, two weeks ago, v0.9.2.2024.05.07.1926

**Reply by jacobsen9026** [reply]
And with the latest release this feature request is included, improvements will come in the form of customization but we're looking good so far.

**Comment by jacobsen9026** [comment]
Customization of email templates is an overarching goal with this discussion. Therefore I will close this discussion and submit an issue for general email template customization.

**Reply by jacobsen9026** [reply]
Issue #447
